src/Auth/AuthController.js

```diff
@@ -1,208 +1,130 @@
-    this.store = getStore();
+  /**
+   * 
+   * @param {*} settings 
+   * @param {*} service 
+   */
+  constructor (settings, service, loginButton) {
     this.settings = settings;
+    validateSettings.call(this, settings);
 
-    // 1. get Language
-    if (settings.authRequest.languageCode != null) {
-      this.store.languageCode = settings.authRequest.languageCode;
+    this.stateChangeListeners = [];
+    if (this.settings.onStateChange) {
+      this.stateChangeListeners.push(this.settings.onStateChange);
     }
-    this.serviceInfoUrl = serviceInfoUrl;
-    this.serviceCustomizations = serviceCustomizations;
-    if (!settings) { throw new Error('settings cannot be null'); }
+    this.service = service;
 
-    try {
-      // -- Check Error CallBack
-      if (this.settings.onStateChange) {
-        this.store.stateChangeListners.push(this.settings.onStateChange);
-      }
+    // probably remove
+    this.languageCode = this.settings.authRequest.languageCode || 'en';    
+    this.messages = Messages(this.languageCode);
 
-      // -- settings 
-      if (!this.settings.authRequest) { throw new Error('Missing settings.authRequest'); }
+    this.loginButton = loginButton;
 
+    function validateSettings (settings) {
+      if (!settings) { throw new Error('settings cannot be null'); }
+      // -- settings 
+      if (!settings.authRequest) { throw new Error('Missing settings.authRequest'); }
+  
       // -- Extract returnURL 
-      this.settings.authRequest.returnURL =
-        this.getReturnURL(this.settings.authRequest.returnURL);
-
-      if (!this.settings.authRequest.requestingAppId) {
+      settings.authRequest.returnURL =
+        this.getReturnURL(settings.authRequest.returnURL);
+  
+      if (!settings.authRequest.requestingAppId) {
         throw new Error('Missing settings.authRequest.requestingAppId');
       }
-      if (!this.settings.authRequest.requestedPermissions) {
+      if (!settings.authRequest.requestedPermissions) {
         throw new Error('Missing settings.authRequest.requestedPermissions');
       }
-
-    } catch (e) {
-      this.store.setState({
-        id: AuthStates.ERROR, message: 'During initialization', error: e
-      });
-      throw (e);
     }
   }
 
   /**
+   * async function to call right after instanciating object
+   * 
    * @returns {PryvService}
    */
   async init () {
-    this.store.setState({ id: AuthStates.LOADING });
-
-    await this.fetchServiceInfo();
-    this.checkAutoLogin();
-
-    if (this.store.state.id !== AuthStates.AUTHORIZED) {
-      // 5. Propose Login
-      await this.prepareForLogin();
+    this.serviceInfo = this.service.infoSync();
+    this.state = { status: AuthStates.LOADING };
+    this.assets = await loadAssets(this);
+    
+    const loginButton = this.loginButton;
+    // initialize human interaction interface
+    if (loginButton != null) {
+      this.stateChangeListeners.push(loginButton.onStateChange.bind(loginButton));
```

Comment:
how Login Button would look like in react side - would it be a class or a function? I understood that Perki voted for a function?

Reply:
don't know yet, but function or class with function is pretty much the same, no?

Reply:
with a class it has dirty component definition like here https://github.com/pryv/app-react-native-example/blob/c629512ff84f88f110cefb588fdeffccdef05050/PryvReactNative/views/auth/login-method-selection.js#L18 , works, but not sure if it is nice